api/ble/README.md

```diff
@@ -0,0 +1,305 @@
+| enabled   | boolean | no | false | whether Bluetooth and this device is enabled |
+| services  | array of Service objects | no | [] | Primary services on this device |
+
+<a name="bd_addr">
+The `address` property is a 48 bit value (BD_ADDR) identifying a Bluetooth Smart device. On the UI level the Bluetooth address MUST be represented as 12 hexadecimal characters, possibly divided into sub-parts separated by ‘:’ (e.g., ‘000C3E3A4B69’ or ‘00:0C:3E:3A:4B:69’). On the UI level any number MUST have the most significant bit on left ordering.
+
+<a name="bd_addr_type">
+The `addressType` property can take the following values: `"public"`, `"static-random"`, `"private-resolvable"`, `"private"`.
+
+<a name="bd_name">
+The `name` property represents the Bluetooth device name that a Bluetooth device
+exposes to remote devices. It and can be up to 248 bytes encoded in UTF-8, therefore on the UI level it may be restricted to as few as 62 characters if codepoints outside the range U+0000 to U+007F are used. A device cannot expect that a general remote device is able to handle more than the first 40 characters of the Bluetooth device name. If a remote device has limited display capabilities, it may use only the first 20 characters.
+
+<a name="bd_enabled">
+The `enabled` property MUST be set to `true` if the Bluetooth Smart and the device is enabled, and otherwise to `false`.
+
+<a name="bd_services">
+The `services` property is a read-only array (sequence) of [`BluetoothService`](#service) objects that represents the primary GATT services on this device. A primary service is visible at device root level, and it may contain other services.
+
+#### `BluetoothPeripheralDevice` events
+
+| Event name        | Event callback argument |
+| --------------    | ----------------------- |
+| *enabledchange*   | N/A                     |
+| *connect*         | String (client Bluetooth Device Address) |
+| *disconnect*      | String (client Bluetooth Device Address) |
+| *error*           | Error                   |
+
+<a name="onenabledchange"></a>
+The `enabledchange` event is emitted when the [`enabled`](#bd_enabled) property is changed. Listeners should check the value of the `enabled` property.
+
+<a name="onconnect"></a>
+The `connect` event is emitted when a Bluetooth Smart device in Central mode (client) is connected. Listeners are invoked with the client's [Bluetooth Device Address](#bd_addr) as a string argument.
+
+<a name="ondisconnect"></a>
+The `disconnect` event is emitted when a connected client device is disconnected. Listeners are invoked with the client's [Bluetooth Device Address](#bd_addr) as a string argument.
+
+<a name="onerror"></a>
+The `error` event is emitted when a Bluetooth error needs to be reported to the application. Listeners are invoked with an [`Error`](https://nodejs.org/api/errors.html#errors_class_error) object as argument.
+
+#### `BluetoothPeripheralDevice` methods
+
+<a name="enable"></a>
+##### The `enable()` method
```

Comment:
You say "request" but don't talk about what happens if it fails... does it throw? reject promiose?